crates/hyperqueue/src/dashboard/ui/screens/auto_allocator/screen.rs

```diff
@@ -0,0 +1,178 @@
+use std::default::Default;
+use std::time::SystemTime;
+use termion::event::Key;
+
+use crate::dashboard::ui::screen::Screen;
+use crate::dashboard::ui::styles::{style_footer, style_header_text};
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::text::draw_text;
+
+use crate::dashboard::data::alloc_timeline::AllocationQueueInfo;
+use crate::dashboard::data::DashboardData;
+use crate::dashboard::ui::screen::controller::ScreenController;
+use crate::dashboard::ui::screens::auto_allocator::alloc_timeline_chart::AllocationsChart;
+use crate::dashboard::ui::screens::auto_allocator::allocations_info_table::AllocationInfoTable;
+use crate::dashboard::ui::screens::auto_allocator::queue_info_table::AllocationQueueInfoTable;
+use crate::dashboard::ui::screens::auto_allocator::queue_params_display::QueueParamsTable;
+use crate::server::autoalloc::DescriptorId;
+use tui::layout::{Constraint, Direction, Layout, Rect};
+
+#[derive(Default)]
+pub struct AutoAllocatorScreen {
+    queue_info_table: AllocationQueueInfoTable,
+    queue_params_table: QueueParamsTable,
+    allocations_info_table: AllocationInfoTable,
+    allocations_chart: AllocationsChart,
+
+    component_in_focus: FocussedComponent,
+}
+
+enum FocussedComponent {
+    QueueParamsTable,
+    AllocationInfoTable,
+}
+
+impl Screen for AutoAllocatorScreen {
+    fn draw(&mut self, frame: &mut DashboardFrame) {
+        let layout = AutoAllocScreenLayout::new(frame);
+        draw_text(
+            "AutoAlloc Info",
+            layout.header_chunk,
+            frame,
+            style_header_text(),
+        );
+
+        match self.component_in_focus {
```

Comment:
Let's remove the duplication from this, first create the booleans (styles) and then pass it to the `draw` functions.